src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -214,25 +214,26 @@ private static uint HandleEventPeerCertificateReceived(State state, ref Connecti
 
             try
             {
-                if (OperatingSystem.IsWindows())
+                if (connectionEvent.Data.PeerCertificateReceived.PlatformCertificateChainHandle != IntPtr.Zero)
```

Comment:
Is this right? Is there any relation between `PlatformCertificateChainHandle` and `PlatformCertificateHandle`? Like that if the first is not null then the other won't be as well?

Because originally you had different null-check conditions there.

Reply:
good catch. In general they should be both set or empty but it is strange to test one filed and then use other. I did yet another attempt to refactor and make that more clear.

For Unix path, the PlatformCertificateHandle points to DER encoded certificate while PlatformCertificateChainHandle points to PKCS7 container containing possibly list of certificates. (the format allows additional parts but that is not generated by msquic) 

On Windows the intermediate CA cache is maintained by OS so we don't need to bother. 
On Linux is everything is trusted and intermediates present we could simple grab the leaf cert.
But we don't know that so we attempt to get as many certs provided by the peer to improve our chances that X509Chain con construct full chain.